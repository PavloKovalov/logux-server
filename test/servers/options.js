#!/usr/bin/env node
var fs = require('fs')
var path = require('path')
var Server = require('../../server')

var app = new Server({
  env: 'test',
  nodeId: 'server',
  subprotocol: '1.0.0',
  supports: '1.x'
})

var defaults = {
  port: '1337',
  host: '127.0.0.1',
  key: fs.readFileSync(path.join(__dirname, '../fixtures/key.pem')),
  cert: fs.readFileSync(path.join(__dirname, '../fixtures/cert.pem'))
}

app.auth(function () {
  return Promise.resolve(true)
})

var options = app.loadOptions(process, defaults)
app.listen(options)

setTimeout(function () {
  app.destroy()
}, 10)
