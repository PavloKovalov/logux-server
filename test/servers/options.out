 INFO   Logux server is listening at 1970-01-01 00:00:00        Logux server: 0.1.1        PID:          21384        Node ID:      server        Environment:  test        Subprotocol:  1.0.0        Supports:     1.x        Listen:       wss://127.0.0.1:31337

 INFO   Shutting down Logux server at 1970-01-01 00:00:00

